plugins/docker/docker_

```diff
@@ -0,0 +1,198 @@
+in the directory /etc/munin/plugin-conf.d/ with a config like this:
+
+=over 2
+
+    [docker_*]
+    user root
+    env.DOCKER_HOST unix://var/run/docker.sock
+
+=back
+"""
+
+import os
+import sys
+import docker
+
+
+def getContainersStatus(client):
+    running = 0
+    paused = 0
+    created = 0
+    restarting = 0
+    removing = 0
+    exited = 0
+    dead = 0
+    for container in client.containers.list():
+        if container.status == 'running':
+            running += 1
+        elif container.status == 'paused':
+            paused += 1
+        elif container.status == 'created':
+            created += 1
+        elif container.status == 'restarting':
+            restarting += 1
+        elif container.status == 'removing':
+            removing += 1
+        elif container.status == 'exited':
+            exited += 1
+        elif container.status == 'dead':
+            dead += 1
+    print('running.value', running)
+    print('paused.value', paused)
+    print('created.value', created)
+    print('restarting.value', restarting)
+    print('removing.value', removing)
+    print('exited.value', exited)
+    print('dead.value', dead)
+
+
+def getContainersCpu(client):
+    for container in client.containers.list():
+        stats = container.stats(stream=False)
+        cpu_count = len(stats["cpu_stats"]["cpu_usage"]["percpu_usage"])
+        cpu_percent = 0.0
+        cpu_delta = float(stats["cpu_stats"]["cpu_usage"]["total_usage"]) \
+            - float(stats["precpu_stats"]["cpu_usage"]["total_usage"])
+        system_delta = float(stats["cpu_stats"]["system_cpu_usage"]) \
+            - float(stats["precpu_stats"]["system_cpu_usage"])
+        if system_delta > 0.0:
+            cpu_percent = cpu_delta / system_delta * 100.0 * cpu_count
+        print(container.name + '.value', cpu_percent)
+
+
+def getContainersMemory(client):
+    for container in client.containers.list():
+        stats = container.stats(stream=False)['memory_stats']
+        print(container.name + '.value', stats['stats']['total_rss'])
+
+
+def main():
+    try:
+        mode = sys.argv[1]
+    except IndexError:
+        mode = ""
+    wildcard = sys.argv[0].split("docker_")[1].split("_")[0]
+
+    if mode == "suggest":
+        print("containers")
+        print("cpu")
+        print("images")
+        print("memory")
+        print("status")
+        print("volumes")
+        sys.exit(0)
+
+    client = docker.from_env()
+
+    if wildcard == "status":
+        if mode == "config":
+            print("graph_title Docker status")
+            print("graph_vlabel containers")
+            print("graph_category virtualization")
+            print("running.label RUNNING")
+            print("paused.label PAUSED")
+            print("created.label CREATED")
+            print("restarting.label RESTARTING")
+            print("removing.label REMOVING")
+            print("exited.label EXITED")
+            print("dead.label DEAD")
+            sys.exit(0)
```

Comment:
I think, it would improve the readability of the flow control, if you simply remove the `exit` calls.

At the moment `exit` is called after every configuration branch, but not after printing the values. There is not difference in practice, but it feels a bit confusing for me.

Reply:
Of course, I just removed that!